cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -88,17 +88,17 @@ func (c *CStorVolumeReplicaController) cVREventHandler(operation common.QueueOpe
 	case common.QOpAdd:
 		glog.Infof("Processing cvr added event: %v, %v", cVR.ObjectMeta.Name, string(cVR.GetUID()))
 
-		importedFlag := <-common.IsImported
-		if importedFlag {
+		common.Mux.Lock()
+		if common.IsImported {
+			common.Mux.Unlock()
 			// To check if volume is already imported with pool.
-			importedFlag = common.CheckForInitialImportedPoolVol(common.InitialImportedPoolVol, fullVolName)
+			importedFlag := common.CheckForInitialImportedPoolVol(common.InitialImportedPoolVol, fullVolName)
 			if importedFlag && !IsInitStatus(cVR) {
 				glog.Infof("CStorVolumeReplica %v is already imported", string(cVR.ObjectMeta.UID))
 				c.recorder.Event(cVR, corev1.EventTypeNormal, string(common.SuccessImported), string(common.MessageResourceImported))
 				return string(apis.CVRStatusOnline), nil
 			}
 		}
-
```

Comment:
The below code checks for a rogue case where an ADD event is received for the same file-system volume name but different custom-resource name. This needs to be addressed by having a status like ERRORDUPLICATE.

Reply:
will add